File: Android App/app/src/main/java/revert/shredsheets/models/scales/MelodicMinor.java
package revert.shredsheets.models.scales;

public class MelodicMinor extends Scale {

    @Override
    public int[] getScaleIntervals() {
        return new int[]{ 2, 1, 2, 2, 2, 2, 1};
    }

    @Override
    public String getName() {
        return "Melodic Minor";
    }

    @Override
    public String[] getModeNames() {
        return new String[]{"Jazz Minor", "Dorian b9", "Lydian Augmented", "Lydian Dominant", "Mixolydian b6", "SemiLocrian", "SuperLocrian"};
    }
}

File: Shredsheets 3/core/src/com/bm/shredsheets/models/scales/MelodicMinor.kt
package com.bm.shredsheets.models.scales

class MelodicMinor : Scale() {
    override val scaleIntervals: IntArray
        get() = intArrayOf(2, 1, 2, 2, 2, 2, 1)

    override val name: String
        get() = "Melodic Minor"

    override fun getModeNames(): Array<String> {
        return arrayOf("Jazz Minor", "Dorian ♭9", "Lydian Augmented", "Lydian Dominant", "Mixolydian ♭6", "SemiLocrian", "SuperLocrian")
    }

    override val getEnum: Scales
        get() = Scales.MelodicMinor

    override val degrees: Array<IntArray>
        get() = arrayOf(
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7))

    override val chords: Array<String>
        get() = arrayOf("mΔ7", "m7", "M7 ♭5", "7", "7", "m7 ♭5", "m7 ♭5")
}